packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,217 +73,375 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-
-    let args = {
-        let mut a = Vec::new();
-        for arg in args {
-            a.push(arg.clone());
-        }
-        a
-    };
-    let args = mv8::Values::from_vec(args);
-    let imported = f
-        .call(args)
-        .map_err(|e| Error::FileImport(path.into(), e.into()))?;
+    args: &[v8::Local<'s, v8::Value>],
+) -> JSRunnerResult<v8::Local<'s, v8::Value>> {
+    let v = eval_file(scope, path)?;
+    let f: v8::Local<'_, v8::Function> = v.try_into().map_err(|e| {
+        JSRunnerError::FileImport(path.into(), format!("Failed to wrap file: {:?}, {e}", &v))
+    })?;
+
+    let global_context = context.global(scope);
+    let imported = f.call(scope, global_context.into(), args).ok_or_else(|| {
+        JSRunnerError::FileImport(path.into(), "could not call function".to_string())
+    })?;
 
     Ok(imported)
 }
 
-impl<'m> Embedded<'m> {
-    fn import(mv8: &'m MiniV8) -> Result<Self> {
-        let arrow = eval_file(mv8, "./src/worker/runner/javascript/apache-arrow-bundle.js")?;
-        let hash_stdlib = eval_file(mv8, "./src/worker/runner/javascript/hash_stdlib.js")?;
-        let hash_util = import_file(mv8, "./src/worker/runner/javascript/hash_util.js", vec![
-            &arrow,
-        ])?;
-        let batches_prototype = import_file(mv8, "./src/worker/runner/javascript/batch.js", vec![
-            &arrow, &hash_util,
-        ])?;
-
-        let ctx_import = import_file(mv8, "./src/worker/runner/javascript/context.js", vec![
-            &hash_util,
-        ])?;
-        let ctx_import = ctx_import.as_array().ok_or_else(|| {
-            Error::FileImport(
-                "./src/worker/runner/javascript/context.js".into(),
-                "Couldn't get array (of functions) from 'context.js'".into(),
+impl<'s> Embedded<'s> {
+    fn import(
+        scope: &mut v8::HandleScope<'s>,
+        context: v8::Local<'s, v8::Context>,
+    ) -> JSRunnerResult<Self> {
+        let arrow = eval_file(
+            scope,
+            "./src/worker/runner/javascript/apache-arrow-bundle.js",
+        )?;
+        let hash_stdlib = eval_file(scope, "./src/worker/runner/javascript/hash_stdlib.js")?;
+        let hash_util = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/hash_util.js",
+            &[arrow],
+        )?;
+        let batches_prototype = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/batch.js",
+            &[arrow, hash_util],
+        )?;
+
+        let ctx_import = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/context.js",
+            &[hash_util],
+        )?;
+        let ctx_import: v8::Local<'_, v8::Array> = ctx_import.try_into().map_err(|e| {
+            JSRunnerError::FileImport(
+                "./src/worker/runner/javascript/context.js".to_string(),
+                format!("Couldn't get array (of functions) from 'context.js': {e}"),
```

Comment:
```suggestion
        let ctx_import: v8::Local<'_, v8::Array> = ctx_import.try_into().map_err(|err| {
            JSRunnerError::FileImport(
                "./src/worker/runner/javascript/context.js".to_string(),
                format!("Couldn't get array (of functions) from 'context.js': {err}"),
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/02af15786b564ada4f9992194281ff0a3281afb9